index.mjs

```diff
@@ -1,61 +1,67 @@
-import {spawn as spwn} from 'child_process';
-import fs from 'fs';
-
-const spawn = (cmd, args) => new Promise((resolve, reject) => {
-  const cp = spwn(cmd, args);
-  const error = [];
-  const stdout = [];
-  cp.stdout.on('data', (data) => {
-    stdout.push(data.toString());
-  });
-
-  cp.on('error', (e) => {
-    error.push(e.toString());
-  });
+import { spawn as spwn } from "child_process";
+import fs from "fs";
 
-  cp.on('close', () => {
-    if (error.length) reject(error.join(''));
-    else resolve(stdout.join(''));
+const spawn = (cmd, args) =>
+  new Promise((resolve, reject) => {
+    const cp = spwn(cmd, args, { stdio: "inherit" });
+    cp.on("close", (code) => {
+      if (code === 0) resolve();
+      else reject(new Error("Exit code: " + code));
+    });
   });
-});
 
 function readFile() {
-  return fs.readFileSync('./links.txt').toString()
+  return fs.readFileSync("./links.txt", "utf8");
 }
 
 async function stateMachine(textContent) {
-  // const lines = textContent.split('\n').slice(0, 10)
-  const lines = textContent.split('\n')
+  const lines = textContent.split("\n");
 
   let seasonNumber;
   let episodeNumber;
   let episodeName;
-  let episodeUrl;
-
 
   for (let line of lines) {
-    const cleanedLine = line.trim()
-    const lcLine = cleanedLine.toLocaleLowerCase();
-    if (lcLine.startsWith('sezon')) {
-      seasonNumber = lcLine.split(' ')[1]
-      console.log(`Settings seasonNumber to ${seasonNumber}`);
-    } else if (lcLine.match(/^\d+./)) {
-      [episodeNumber, episodeName] = cleanedLine.split('. ')
-      console.log(episodeNumber, episodeName);
-    } else if (lcLine.startsWith('http')) {
-      episodeUrl = cleanedLine
-      console.log(episodeUrl);
-
-      const directoryName = `Świat.Według.Kiepskich.S${seasonNumber.padStart(2, '0')}`
-      let sanitizedEpisodeName = episodeName.replace(/[^A-Za-z0-9À-ž]/g, '.')
-      if (sanitizedEpisodeName.endsWith('.')) {
-        sanitizedEpisodeName = sanitizedEpisodeName.substring(0, sanitizedEpisodeName.length-1)
-      }
-      const fileName = `Świat.Według.Kiepskich.S${seasonNumber.padStart(2, '0')}E${episodeNumber.padStart(3, '0')}.${sanitizedEpisodeName}`
-      console.log(directoryName,fileName);
-
-      await spawn("yt-dlp", [episodeUrl, '-o', `./downloads/${directoryName}/${fileName}.%(ext)s`])
+    const cleaned = line.trim();
+    const lc = cleaned.toLowerCase();
+
+    if (lc.startsWith("sezon")) {
+      seasonNumber = lc.split(" ")[1];
+      console.log("\n=== SEZON", seasonNumber, "===\n");
+      continue;
+    }
+
+    if (/^\d+\./.test(cleaned)) {
+      const parts = cleaned.split(". ");
+      episodeNumber = parts[0];
+      episodeName = parts[1];
+      continue;
+    }
+
+    if (cleaned.startsWith("http")) {
+      const episodeUrl = cleaned;
+
+      const directoryName = `Świat.Według.Kiepskich.S${seasonNumber.padStart(2, "0")}`;
+      let sanitized = episodeName.replace(/[^A-Za-z0-9À-ž]/g, ".");
```

Comment:
We should be specific about what this variable includes.
Skimming through the code you don't really know what exactly is sanitized and have to focus more to understand it. Having it named `sanitizedEpisodeName` would solve this problem.